### PR TITLE
Generalises multiple buy from market only. Add it to 1870.

### PR DESCRIPTION
**Pull request by daniel-sousa-me**
There was a bug on this feature that allowed players to buy a share from the ipo and then buy multiple from the market. This has also been fixed.

### CODE REVIEW

**Comment by daniel-sousa-me** [comment]
This may pin and/or archive some games

File: lib/engine/step/buy_sell_par_shares.rb [hunk 1]
```diff
@@ -179,10 +183,15 @@ def pass!
...
+      def can_buy_multiple?(_entity, corporation, owner)
+        if @game.multiple_buy_only_from_market?
+          return false unless owner.share_pool?
+          return false if @round.bought_from_ipo
+        end
+
         corporation.buy_multiple? &&
-         @round.current_actions.none? { |x| x.is_a?(Action::Par) } &&
-         @round.current_actions.none? { |x| x.is_a?(Action::BuyShares) && x.bundle.corporation != corporation }
+          @round.current_actions.none? { |x| x.is_a?(Action::Par) } &&
```
Comment (jenf) [inline lib/engine/step/buy_sell_par_shares.rb:193]: stray change? undo.

Comment (daniel-sousa-me) [reply lib/engine/step/buy_sell_par_shares.rb:193]: It was indented with one space instead of two. Is that right?

Comment (jenf) [reply lib/engine/step/buy_sell_par_shares.rb:193]: yep, it's unchanged

**Comment by jenf** [comment]
1830 games broken & rules broken
https://18xx.games/game/27054?action=567 (rules broken)
https://18xx.games/game/30631?action=361
https://18xx.games/game/30721?action=192

**Comment by jenf** [comment]
All 1882 games are fine... lgtm.

**Review by jenf — APPROVED** [review]

**Comment by dfannius** [comment]
The 1882 rules explicitly allow the old behavior (buy any number of brown shares from anywhere), so if any 1882 games had to be pinned, that would be a sign of a bug. So it's a good thing that they're all fine.

**Comment by jenf** [comment]
yep, that's why i ran them :)

**Pull request merged**